src/app/views/panels/scales_panel.tsx

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+import * as React from "react";
+import * as R from "../../resources";
+
+import { EventSubscription, Prototypes } from "../../../core";
+import { SVGImageIcon } from "../../components";
+
+import { AppStore } from "../../stores";
+import { ReorderListView } from "./object_list_editor";
+import { ContextedComponent } from "../../context_component";
+import {
+  ScaleMapping,
+  Glyph,
+  ChartElement,
+  ObjectProperties,
+  Element,
+  Scale
+} from "../../../core/specification";
+
+function getObjectIcon(classID: string) {
+  return R.getSVGIcon(
+    Prototypes.ObjectClasses.GetMetadata(classID).iconPath || "object"
+  );
+}
+
+export class ScalesPanel extends ContextedComponent<
+  {
+    store: AppStore;
+  },
+  {}
+> {
+  public mappingButton: Element;
+  private tokens: EventSubscription[];
+
+  public componentDidMount() {
+    this.tokens = [];
+    this.tokens.push(
```

Comment:
Nitpick:  This could be
```
this.tokens = [
   this.store.addListener(AppStore.EVENT_GRAPHICS, () => this.forceUpdate()),
   this.store.addListener(AppStore.EVENT_SELECTION, () => this.forceUpdate()),
   this.store.addListener(AppStore.EVENT_SAVECHART, () => this.forceUpdate())
]
```

Reply:
Fixed